src/rust/engine/fs/store/src/snapshot.rs

```diff
@@ -99,13 +83,10 @@ impl Snapshot {
   >(
     store: Store,
     file_digester: S,
-    path_stats: &[PathStat],
+    path_stats: Vec<PathStat>,
   ) -> Result<Digest, String> {
-    let mut sorted_path_stats = path_stats.to_owned();
-    #[allow(clippy::unnecessary_sort_by)]
-    sorted_path_stats.sort_by(|a, b| a.path().cmp(b.path()));
-    Snapshot::ingest_directory_from_sorted_path_stats(store, file_digester, &sorted_path_stats)
-      .await
+    let path_stats = PathStat::normalize_path_stats(path_stats)?;
```

Comment:
Ditto that `digest_from_path_stats()` used to not check for duplicates. It now does, and I think this is the correct behavior as we aren't certain what the input `path_stats` we're getting are like.